release/rancher/rancher.go

```diff
@@ -296,3 +335,144 @@ func createPRFromRancher(ctx context.Context, rancherBaseBranch, title, branchNa
 
 	return err
 }
+
+type ContentLine struct {
+	Line    int
+	File    string
+	Content string
+}
+
+type Content struct {
+	RancherImages  []ContentLine
+	FilesWithRC    []ContentLine
+	MinFilesWithRC []ContentLine
+	ChartsWithDev  []ContentLine
+	KDMWithDev     []ContentLine
+}
+
+func CheckRancherRCDeps(forCi bool, org, repo, commitHash, files string) error {
+	var (
+		matchCommitMessage bool
+		badFiles           bool
+		content            Content
+	)
+
+	devDependencyPattern := regexp.MustCompile(`dev-v[0-9]+\.[0-9]+`)
+	rcTagPattern := regexp.MustCompile(`-rc[0-9]+`)
+
+	httpClient := ecmHTTP.NewClient(time.Second * 15)
+
+	if repo == "" {
+		repo = rancherRepo
+	}
+	if org == "" {
+		org = rancherOrg
+	}
+
+	//should return data if executed in rancher project root path
+	err := writeRancherImagesDeps(&content, rcTagPattern)
+	if err != nil {
+		return err
+	}
+
+	for _, filePath := range strings.Split(files, ",") {
+		repoContent, err := repository.ContentByFileNameAndCommit(org, repo, commitHash, filePath, &httpClient)
+		if err != nil {
+			return err
+		}
+
+		scanner := bufio.NewScanner(strings.NewReader(string(repoContent)))
+		lineNum := 1
+
+		for scanner.Scan() {
+			lineByte, line := formatLineByte(scanner.Text())
+
+			if devDependencyPattern.Match(lineByte) {
+				badFiles = true
+				lineContent := ContentLine{File: filePath, Line: lineNum, Content: formatContentLine(line)}
+				lineContentLower := strings.ToLower(lineContent.Content)
+				if strings.Contains(lineContentLower, "chart") {
+					content.ChartsWithDev = append(content.ChartsWithDev, lineContent)
+				}
+				if strings.Contains(lineContentLower, "kdm") {
+					content.KDMWithDev = append(content.KDMWithDev, lineContent)
+				}
+			}
+			if strings.Contains(filePath, "/package/Dockerfile") {
+				if !strings.Contains(line, "_VERSION") {
+					continue
+				}
+				matches := regexp.MustCompile(`CATTLE_(\S+)_MIN_VERSION`).FindStringSubmatch(line)
+				if len(matches) == 2 && strings.Contains(line, "-rc") {
```

Comment:
This condition does not include setting barflies, not sure if that's intentional

Reply:
No, gonna add the flag